ncm-download/src/main/perl/download.pm

```diff
@@ -248,11 +249,29 @@ sub get_remote_timestamp
     return $head->headers->last_modified();
 }
 
+sub _cleanup_tmpfile
+{
+    my ($self, $tmpfn) = @_;
+    if ($self->file_exists($tmpfn)) {
+        unless (unlink($tmpfn)) {
+            $self->error("Unable to delete temporary file $tmpfn: $!");
+            return 0;
+        }
+    }
+    return 1;
+}
+
+sub _tempfile
+{
+    my ($self, $fn) = @_;
+    return dirname($fn) . "/." . basename($fn) . ".part";
```

Comment:
add a comment that the tempfile must be in the same dir as the destination (so the move/rename is atomic)
it's to prevent other devs from causing serious issues when they want to rename the tempfile to something else